src/CommunityToolkit.Maui/Behaviors/AnimationBehavior.shared.cs

```diff
@@ -76,31 +75,42 @@ protected override void OnDetachingFrom(VisualElement bindable)
 	/// <inheritdoc/>
 	protected override async void OnTriggerHandled(object? sender = null, object? eventArgs = null)
 	{
-		await OnAnimate();
+		await OnAnimate(CancellationToken.None);
 
 		base.OnTriggerHandled(sender, eventArgs);
 	}
 
-	static object CreateAnimateCommand(BindableObject bindable)
-		=> new Command(async () => await ((AnimationBehavior)bindable).OnAnimate().ConfigureAwait(false));
+	static Command<CancellationToken> CreateAnimateCommand(BindableObject bindable)
+	{
+		var animationBehavior = (AnimationBehavior)bindable;
+		return new Command<CancellationToken>(async token => await animationBehavior.OnAnimate(token).ConfigureAwait(false));
+	}
+
+	static void OnAnimateCommandChanging(BindableObject bindable, object oldValue, object newValue)
+	{
+		if (newValue is not Command<CancellationToken>)
+		{
+			var animationBehavior = (AnimationBehavior)bindable;
+			throw new InvalidOperationException($"{nameof(AnimateCommand)} must of Type {animationBehavior.AnimateCommand.GetType().FullName}");
+		}
+	}
 
-	Task OnAnimate()
+	async Task OnAnimate(CancellationToken token)
 	{
 		if (View is null || AnimationType is null)
 		{
-			return Task.CompletedTask;
+			return;
 		}
 
 		View.CancelAnimations();
 
 		try
 		{
-			return AnimationType.Animate(View);
+			await AnimationType.Animate(View, token);
```

Comment:
```suggestion
			// We must `await` `AnimationType.Animate()` here in order to properly implement `Options.ShouldSuppressExceptionsInAnimations`
			// Returning the `Task` would cause the `OnAnimate()` method to return immediately, before `AnimationType.Animate()` has completed. Returning immediately exits our try/catch block and thus negates our opportunity to handle any Exceptions which breaks `Options.ShouldSuppressExceptionsInAnimations`.
			await AnimationType.Animate(View, token);
```